src/ledger.rs

```diff
@@ -417,33 +427,100 @@ impl LedgerWriter {
 #[derive(Debug)]
 pub struct LedgerReader {
     data: BufReader<File>,
+    file_name: String,
+    tailing: bool,
+    pub cur_pos: usize,
+    pub last_len: usize,
 }
 
 impl Iterator for LedgerReader {
     type Item = io::Result<Entry>;
 
     fn next(&mut self) -> Option<io::Result<Entry>> {
-        match next_entry(&mut self.data) {
-            Ok(entry) => Some(Ok(entry)),
-            Err(_) => None,
+        if !self.tailing {
+            return match next_entry(&mut self.data, &mut self.cur_pos) {
+                Ok(entry) => Some(Ok(entry)),
+                Err(_) => None,
+            };
+        }
+
+        let entry_result;
+
+        loop {
```

Comment:
This doesn't look like a typical implementation of an iterator. Does this make `next` an `O(n)` operation?

Reply:
How about we explore that landscape on our end? I'll sync up with some folks on Monday and circle back here.

Reply:
Sounds like a plan -- in any case, I can use this branch for demo purposes for some amount of time while building out the UI. Thank you in advance for your help!